tests/libnet/ping.go

```diff
@@ -50,7 +50,13 @@ func PingFromVMConsole(vmi *v1.VirtualMachineInstance, ipAddr string, args ...st
 
 	err := console.RunCommand(vmi, cmdCheck, maxCommandTimeout)
 	if err != nil {
-		return fmt.Errorf("Failed to ping VMI %s, error: %v", vmi.Name, err)
+		return fmt.Errorf("Failed to ping VMI %s with status %+v, error: %v", vmi.Name, vmi.Status, err)
 	}
 	return nil
 }
+
+func PingFromVMConsoleFor(vmi *v1.VirtualMachineInstance, ipAddr string, args ...string) func() error {
```

Comment:
I have hard time to understand the name.
Wrapping using a lambda at the usage may be simpler.

An alternative is to embed the retry in the ping itself with a non-eventually version.